File: backend/models/investor.go
package models

import "gorm.io/gorm"

type InvestorType string

const (
	TypeRUPS               InvestorType = "rups"
	TypeAnnualReport       InvestorType = "annual-report"
	TypeProspectus         InvestorType = "prospectus"
	TypeFinancialStatement InvestorType = "financial-statement"
)

type InvestorRelations struct {
	gorm.Model
	Year    string       `json:"year" gorm:"type:varchar(4);not null"`
	Quartal string       `json:"quartal" gorm:"type:varchar(10)"`
	URL     string       `json:"url" gorm:"type:text;not null"`
	Type    InvestorType `json:"type" gorm:"type:text;"`
}

File: frontend/services/investor.ts
import apiClient from "@/services/api";
import { GlobalResponse } from "@/services/type/responses";

export type InvestorRelationType = 'rups' | 'annual-report' | 'prospectus' | 'financial-statement'

export type InvestorRelationData = {
    ID?: number;
    year: string;
    quartal?: string;
    url: string;
    type: InvestorRelationType;
}

export const uploadInvestorRelationData = async (payload: InvestorRelationData) => {
    return apiClient.post<GlobalResponse<null>>(`/api/investor-relation/`, payload);
}

export const getInvestorRelationData = async (type?: string) => {
    return apiClient.get<GlobalResponse<InvestorRelationData[]>>(`/api/investor-relation/?type=${type ? type: ''}`);
}
